pkg/objstore/client/factory.go

```diff
@@ -81,5 +85,19 @@ func NewBucket(logger log.Logger, confContentYaml []byte, reg prometheus.Registe
 	if err != nil {
 		return nil, errors.Wrap(err, fmt.Sprintf("create %s client", bucketConf.Type))
 	}
-	return objstore.NewTracingBucket(objstore.BucketWithMetrics(bucket.Name(), bucket, reg)), nil
+
+	var prefixedBucket objstore.Bucket
+	if validPrefix(bucketConf.Config.Prefix) {
+		prefixedBucket = objstore.NewPrefixedBucket(bucket, bucketConf.Config.Prefix)
+		level.Debug(logger).Log("msg", "using prefix on bucket access", "prefix", bucketConf.Config.Prefix)
+	} else {
+		prefixedBucket = bucket
```

Comment:
I think this else block is unnecessary. What about just using the code below on line 91.
``` go
bucket = objstore.NewPrefixedBucket(bucket, bucketConf.Config.Prefix)
```

Reply:
The thing is that we wanted to not use the prefixedBucket decorator in cases it is not needed. That would reduce the blast radius in case we introduce a bug. What do you think?

Reply:
nice idea @GiedriusS, we moved it to NewPrefixedBucket